docs/tutorial.rst

```diff
@@ -0,0 +1,1842 @@
+.. currentmodule:: reframe.core.pipeline.RegressionTest
+
+================
+ReFrame Tutorial
+================
+
+This tutorial will cover the basic concepts of ReFrame and will get you started with the framework.
+For more specific topics, you should refer to ":doc:`howto`" as well as to the ":doc:`topics`" for an in-depth understanding of some of the framework's concepts.
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 3
+
+Requirements
+============
+
+To run this tutorial you need ``docker`` for the local examples and ``docker compose`` for the examples emulating a Slurm cluster.
+
+The tutorial container images provide already the latest ReFrame version installed.
+For installing a stand-alone version of ReFrame, please refer to the ":doc:`started`" guide.
+
+All tutorial examples are located under the ``reframe-examples`` directory inside the container's working directory.
+
+
+Running the local examples
+--------------------------
+
+To run the local examples, launch the single-node tutorial container by binding mounting the examples:
+
+.. code-block:: bash
+
+   docker run -h myhost -it --mount type=bind,source=(pwd)/examples/,target=/home/user/reframe-examples reframe-tut-singlenode:latest /bin/bash
```

Comment:
May consider mentioning navigation to RFM's project root, since these paths assume that working directory. Missing a `$` character here - should be `$(pwd)/examples,...`

This didn't run out of the box for me. Is there anything I need to do in order to get access to these images?

Reply:
discussion: `reframe-tut-singlenode` here has been built from a dockerfile locally, not pulled from a registry. Lowest effort approach would be providing the dockerfile in-repo (requires cloning), and adding a `docker build` step to the tutorial. HIgher effort approach would be setting up GH actions automation alongside RFM releases to push an updated image to a registry, which would allow for using the tutorial containers without cloning.

Reply:
Let's go for the build+run solution at the moment, which is tested :)